server/tests/common/backend.py

```diff
@@ -68,6 +69,35 @@ def app(backend: Backend, monkeypatch: pytest.MonkeyPatch) -> Iterator[AsgiApp]:
     asgi_app.state.backend = backend
     yield asgi_app
 
+@contextmanager
+def run_asgi_server(app: AsgiApp) -> Iterator[tuple[str, int]]:
```

Comment:
I'd rather move this code in the `test_sse` file.
This is because this function is not good for arbitrary use (see my comment about the process fork making `app` & `backend` variables no longer usable)

As matter of fact I think we should add this info in the function name given it's a very unexpected behavior (something like `fork_process_to_run_asgi_server`)

Reply:
done